### DISCUSSION TITLE
How to host tracking script.js on a CDN?

### DISCUSSION THREAD

**Discussion by rhbkweb** [umami-software/umami#2237] [Q&A]
Hi,

Just started using umami installed on cloudron.io control panel (uses docker), and it’s awesome!

My objective is for the umami tracking script to have the best worldwide speed and performance, so my idea is to host it on a CDN provider (just the script.js file).

Is it a viable and good option? Any idea on how to do it? I know the basecs of working with a CDN, so I know how to host the script.js there and have a link for it. But do not know what changes are needed on umami.

Thanks

**Comment by aaro-n** [comment]
Look at the documentation:[data-host-url](https://umami.is/docs/tracker-configuration#:~:text=configure%20its%20behavior.-,data%2Dhost%2Durl,-By%20default%2C%20Umami)

**Comment by rhbkweb** [comment]
@aaro-n

Thanks for the link. After reading the docs, it seems the "data-host-url" is the path/url location were umami sends the data. But before sending the data, the script.js must be read/downloaded by the user/client. correct?

So how to have both: get/download the script.js fro CDN and then send the data to the same CDN?

Thanks

**Reply by aaro-n** [reply]
Both `src` and `data-host-url` items can use CDN,
```
   src="//js.123.fun/um/PGmXAF3tDjvN9.js"
   data-website-id="ID"
   data-host-url="https://tg.123.cn"
```
Get the `src` item, visit `https://tg.123.cn/script.js` to save the js file to the desktop, upload `script.js` to S3 or other cloud storage, cloud storage can be bound Your own domain name, accessed through the domain name.
`data-host-url` can also use CDN, but pay attention to whether the passed IP is correct, pay attention to `CLIENT_IP_HEADER` variable.

The umami I deployed is the CDN used.

`src` uses Youpaiyun CDN
`data-host-url` uses aws cloudfront
